app/src/main/java/com/flab/findmycat/ui/list/CatListAdapter.kt

```diff
@@ -0,0 +1,57 @@
+package com.flab.findmycat.ui.list
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.lifecycle.LifecycleOwner
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.flab.findmycat.databinding.CatItemBinding
+import com.flab.findmycat.domain.Cat
+
+class CatClickListener(val clickListener: (id: String) -> Unit) {
+    fun onClick(cat: Cat) = clickListener(cat.id ?: "")
+}
```

Comment:
```suggestion
fun interface CatClickListener {
    fun onClick(cat: Cat)
}
```
`Listener`를 이런식으로 구현하신 이유가 있으실까요?
불필요하게 코드복잡도만 높이는 구현으로 보여집니다

Reply:
수정했습니다~~!

Reply:
수정했습니다!